apps/api/v2/src/ee/event-types/event-types_2024_04_15/event-types.repository.ts

```diff
@@ -64,6 +71,10 @@ export class EventTypesRepository_2024_04_15 {
     return this.dbRead.prisma.eventType.findUnique({ where: { id: eventTypeId } });
   }
 
+  async getEventTypeBySlug(slug: string) {
+    return this.dbRead.prisma.eventType.findFirst({ where: { slug } });
+  }
```

Comment:
An eventType slug is unique in `userId-slug` and  `teamId-slug` pairing. This function would return the first one that it finds, and it could return a team eventType when looking for a user event, and vice-versa (we cannot trust this function's return). 

We should probably have two separate functions, one for fetching teamEvents and other for fetching user events. Both the functions should take an extra input of teamId/userId respectively.